akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/DebuggingDirectivesExamplesSpec.scala

```diff
@@ -95,4 +97,31 @@ class DebuggingDirectivesExamplesSpec extends RoutingSpec {
       responseAs[String] shouldEqual "logged"
     }
   }
+  "logRequestResultWithResponseTime" in {
+
+    def akkaResponseTimeLoggingFunction(loggingAdapter: LoggingAdapter,
+                                        requestTimestamp: Long,
+                                        level: LogLevel = Logging.InfoLevel)(req: HttpRequest)(res: Any): Unit = {
+      val entry = res match {
+        case Complete(resp) =>
+          val responseTimestamp: Long = System.nanoTime
+          val elapsedTime: Long = (responseTimestamp - requestTimestamp) / 1000000
+          val loggingString = s"""Logged Request:${req.method}:${req.uri}:${resp.status}:${elapsedTime}"""
+          LogEntry(loggingString, level)
+        case Rejected(reason) =>
+          LogEntry(s"Rejected Reason: ${reason.mkString(",")}", level)
+      }
+      entry.logTo(loggingAdapter)
+    }
+    def printResponseTime(log: LoggingAdapter) = {
+      val requestTimestamp = System.nanoTime
+      akkaResponseTimeLoggingFunction(log, requestTimestamp)(_)
+    }
+
+    val logResponseTime = DebuggingDirectives.logRequestResult(LoggingMagnet(printResponseTime(_)))
```

Comment:
@ktoso: If I remove the LoggingMagnet here : It gives me an error of TypeMismatch